app.ts

```diff
@@ -0,0 +1,309 @@
+		let headings = Promise.resolve(fetch("/columns").then(res => res.text())); 
+		let records = Promise.resolve(fetch("/records?from="+fromID+"&to="+toID).then(res => res.text()));
+		Promise.all<string>([headings, records]).then((values) => {
+			generateTable(values[0], values[1]);
+		}).catch(err => console.log(err));
+	}, 500);
+});
+
+
+// Function to create table and render in browser
+function generateTable(headingsStr: string, recordsStr: string) {
+	// Instantiate grid table to append innerHTML
+	let tble = new RenderTableHeading(document.querySelector('#table') as HTMLDivElement);
+
+	// Empty the table in the html
+	let emptyTable = document.querySelector('#table') as HTMLDivElement;
+	emptyTable.innerHTML = "";										// clear table to empty html
+
+	// Recreate table with new headings and records data
+	let interfaceHeading: HasFormatMethod;							// variable of type interface used in creating table headings
+	let interfaceRecords: HasFormatMethod;							// variable of type interface used in creating table rows
+	interfaceHeading = new TableHeadingString(headingsStr);			// call method to generate string containing table heading element
+	tble.constructTableHeadings(interfaceHeading);					// call method to render table headings element in browser
+	interfaceRecords = new RenderTableRows(recordsStr);				// call method to generate string containing table row elements
+}
+
+// Function to target an element in html by its id
+function el(s: string) {
+	return document.getElementById(s);
+}
+
+// Function to create the initial table when loading the page for the first time or when the window size changes
+function createInitialPage(numOfRows: number) {
+	// Clear the timeout every time this function is called
+	clearTimeout(clickTimeout);
+
+	// Check if you're at the start or end of the data set, and reset IDs for change in window size
+	if (fromID > (totalNumofRecords-(numOfRows-1))) {
+		fromID = totalNumofRecords-(numOfRows-1);
+		toID = fromID + (numOfRows-1);
+		fromIDElement.innerHTML = fromID.toString();
+		toIDElement.innerHTML = toID.toString();
+	} else {
+		fromID = fromID;
+		toID = fromID + (numOfRows-1);
+		fromIDElement.innerHTML = fromID.toString();
+		toIDElement.innerHTML = toID.toString();
+	}
+
+	//Request for and set the total number of records
+	fetch("/recordCount").then(res => res.text()).then((value) => {
+		numofrecords.innerHTML = value;
+		totalNumofRecords = Number(value) - 1;
+	}).catch(err => console.log(err))
+
+	// Create table using the new set IDs inside a set timeout
+	clickTimeout = setTimeout(function(){
+		// Use resolved promises to fetch data from backend
+		let headings = Promise.resolve(fetch("/columns").then(res => res.text())); 
+		let records = Promise.resolve(fetch("/records?from="+fromID+"&to="+toID).then(res => res.text()));
+		Promise.all<string>([headings, records]).then((values) => {
+			generateTable(values[0], values[1]);
+		}).catch(err => console.log(err));
+	}, 500);
```

Comment:
This code should be replace with the same function that you call at the end of `#rightarrow` and `#leftarrow`